src/containers/ConnectorManagement.jsx

```diff
@@ -167,8 +169,8 @@ export default class ConnectorManagement extends Component {
     // if wrong connector oauth
     if (eventData.origin !== `${this.props.params.connectorSlug}_oauth`) return
     // get account id from localStorage event and remove the listener
-    window.removeEventListener('storage', this.terminateOAuth.bind(this))
     const accountID = eventData.key
+    window.removeEventListener('storage', this.terminateOAuth)
```

Comment:
For now I would keep here since it could change agin for better other oauth connectors compatibility, won't it? But I agree all this stuff should be out of this `ConnectorManagement.jsx`